src/display.rs

```diff
@@ -166,8 +165,14 @@ impl<'a> DisplayByteSlice<'a> {
 
         let pad_right = if let Some(width) = f.width() {
             let string_len = match f.precision() {
-                Some(max) if self.bytes.len() * 2 > (max + 1) / 2 => max,
-                Some(_) | None => self.bytes.len() * 2,
+                #[rustfmt::skip]
+                Some(max) => {
+                    // Odd max width which makes no sense for hex strings, honour max but use even length.
```

Comment:
Is this statement correct? Yesterday I realised that we may want to support parsing integers in which case `0xa` should be parsable, precision and width should allow `format!("{0:.1}"` in that case, right?

Reply:
Ah, yeah, I think if this is supposed to parse integers (which seems to be the case, given that we're supporting the `0x` prefix) then we ought not to require even lengths.

Reply:
Oh boy, the last two hours has show me that parsing hex integers is a whole can of worms. I am starting to see why it was never included in this crate.